sys-fs/cryfs/cryfs-0.10.0.ebuild

```diff
@@ -0,0 +1,93 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+CMAKE_MAKEFILE_GENERATOR=emake
+PYTHON_COMPAT=( python{2_7,3_{5,6,7}} )
+
+inherit cmake-utils python-any-r1 flag-o-matic
+
+DESCRIPTION="Encrypted FUSE filesystem that conceals metadata"
+HOMEPAGE="https://www.cryfs.org/"
+
+SLOT=0
+IUSE="custom-optimization libressl test update-check"
+
+LICENSE="LGPL-3 BSD-2 MIT"
+# cryfs - LGPL-3
+# scrypt - BSD-2
+# spdlog - MIT
+
+if [[ "${PV}" == 9999 ]] ; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/cryfs/cryfs"
+else
+	SRC_URI="https://github.com/cryfs/cryfs/releases/download/${PV}/${P}.tar.xz"
+	KEYWORDS="~amd64 ~arm ~arm64 ~x86"
+	S="${WORKDIR}"
+fi
+
+RDEPEND=">=dev-libs/boost-1.56:=
+	>=dev-libs/crypto++-5.6.3:=
+	net-misc/curl:=
+	>=sys-fs/fuse-2.8.6:=
+	!libressl? ( dev-libs/openssl:0= )
+	libressl? ( dev-libs/libressl:= )"
+DEPEND="${RDEPEND}
+	${PYTHON_DEPS}"
+
+src_prepare() {
+	cmake-utils_src_prepare
+
+	# remove tests that require internet access to comply with Gentoo policy
+	sed -i -e '/CurlHttpClientTest.cpp/d' -e '/FakeHttpClientTest.cpp/d' test/cpp-utils/CMakeLists.txt || die
+
+	# remove non-applicable warning
+	sed -i -e '/WARNING! This is a debug build. Performance might be slow./d' src/cryfs-cli/Cli.cpp || die
+}
+
+src_configure() {
+	# upstream restricts installing files to Release configuration
+	# (CMAKE_BUILD_TYPE does not affect anything else)
+	local CMAKE_BUILD_TYPE
+	local -a mycmakeargs
+	CMAKE_BUILD_TYPE=Release
+	mycmakeargs=(
```

Comment:
local `mycmakeargs` makes line 53 obsolete.

Reply:
If you define a local variable and assign it on the same line, bash assigns the variable first and then marks it as local, so the variable remains set after leaving the function even though any further changes will be local. Try it out with a test script. This is why it's written this way, as with other areas where the `local` builtin is used. This differs from the behavior of some other shells.